### ISSUE TITLE
Scene keeps reloading after a few steps in iOS

### ISSUE THREAD

**Issue by phrxmd** [mozilla/hubs#3332]
**Description**
After a few steps into a scene in Safari on iOS, I get thrown out of the scene and browser reloads the entire scene

**To Reproduce**
1. Go to https://hubs.mozilla.com/WhkHCsq/living-memory-vr on Safari on iOS 14.1 (tested in iPhone 11 Pro)
2. Move a bit in the direction of the arrows (or any direction, really)
3. You get thrown out of the scene, the entire scene reloads and Safari re-asks you for permission to use the microphone

**Expected behavior**
I should be able to stay in the scene and move through it.

**Screenshots**

**Hardware** 
 - Device: iPhone 11 Pro
 - OS: iOS 14.1
 - Browser: Safari

**Comment by misslivirose** [comment]
I had this happen to me in a separate room today, though it didn't happen immediately on a subsequent reload and doesn't seem to reproduce 100%. It is possible that this is the result from hitting memory limits on iOS and having Safari refresh automatically, but I'm not positive how to confirm that.

**Comment by vincentfretin** [comment]
Yeah it's probably a memory issue. I had a similar issue on a A-Frame project with iPad 2018 iOS 13.7 Safari with only a sphere with a 360 image and a single model with 100k triangles.

**Comment by vincentfretin** [comment]
Your scene is quite heavy 133 MB loaded in 3 minutes with my connection :D
`/vrstats` gives 2800k triangles :scream: when you look at the building.
A memory profile in chrome gives 305 MB of memory. This is probably too much on mobile.

**Comment by vincentfretin** [comment]
In my A-Frame project where I enabled the `stats` component on the `<a-scene>` tag, I can give you some numbers,
- 36 textures, 9 programs, 30 geometries, 151k triangles, 89 MB memory used is ok
- 39 textures sometimes ok
- 41 textures crashed on iPad 2018 6th gen

so the crash seems to be a constraint on the gpu memory I think.

Anyway you should first optimize the scene, 2800k triangles to render is too much. Even for Oculus Quest you should target max 750k, see https://developer.oculus.com/documentation/unity/unity-perf/

**Comment by vincentfretin** [comment]
Well in my case, I say texture, but it's probably related to the load of an additional 100k triangles geometry in the gpu.
I think it's a similar issue in your scene. You can create a new scene without the buildings to see if it still crashes.

**Comment by brianpeiris** [comment]
This does sound like a memory limit issue. Thanks for helping diagnose it, Vincent.
I'm not entirely sure about this, but it's possible that there's little we can do here except maybe display a message to scene creators or viewers that the browser might have crashed due to memory issues. Might be difficult to detect that scenario though.
